File: Data Structures & Algorithms/reverse-linked-list-ii/submission-0.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

 // [1, 2, 3, 4, 5, 6] l: 2 r: 4
 // [1, 4, 3, 2, 5, 6]
public class Solution {
    public ListNode ReverseBetween(ListNode head, int left, int right) {
        if (head == null || left == right) {
            return head;
        }

        ListNode dummyHead = new(0);
        dummyHead.next = head;

        ListNode prev = dummyHead;
        for (int i = 0; i < left - 1; i++) {
            prev = prev.next;
        }

        ListNode curr = prev.next;
        for (int i = 0; i < right - left; i++) {
            ListNode rNode = curr.next;
            curr.next = rNode.next;
            rNode.next = prev.next;
            prev.next = rNode;
        }

        return dummyHead.next;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left = 0
        right = len(heights)-1
        maxWater = 0

        while left < right:
            if min(heights[left], heights[right]) * (right - left) > maxWater:
                print(left, right)
                maxWater = min(heights[left], heights[right]) * (right - left)
            
            if heights[left] > heights[right]:
                right-=1
            else:
                left+=1
        
        return maxWater